client/state/posts/actions.js

```diff
@@ -168,7 +171,59 @@ export function resetPostEdits( siteId, postId ) {
 
 /**
  * Returns an action thunk which, when dispatched, triggers a network request
- * to delete the specified post.
+ * to save the specified post object.
+ *
+ * @param  {Object}   post   Post attributes
+ * @param  {Number}   siteId Site ID
+ * @param  {Number}   postId Post ID
+ * @return {Function}        Action thunk
+ */
+export function savePost( post, siteId, postId ) {
+	return async ( dispatch ) => {
+		dispatch( {
+			type: POST_SAVE,
+			siteId,
+			postId,
+			post
+		} );
+
+		let postHandle = wpcom.site( siteId ).post( postId );
+		postHandle = postHandle[ postId ? 'update' : 'add' ].bind( postHandle );
+		return postHandle( { apiVersion: '1.2' }, post ).then( ( savedPost ) => {
+			dispatch( {
+				type: POST_SAVE_SUCCESS,
+				siteId,
+				postId,
+				savedPost
+			} );
+			dispatch( receivePost( savedPost ) );
+		} ).catch( ( error ) => {
+			dispatch( {
+				type: POST_SAVE_FAILURE,
+				siteId,
+				postId,
+				error
+			} );
+		} );
+	};
+}
+
+/**
+ * Returns an action thunk which, when dispatched, triggers a network request
+ * to trash the specified post.
+ *
+ * @param  {Number}   siteId Site ID
+ * @param  {Number}   postId Post ID
+ * @return {Function}        Action thunk
+ */
+export function trashPost( siteId, postId ) {
```

Comment:
>Would it be worth adding a null check here for `postId` and only calling `savePost` if it is set?

I think it depends. In general I've started shying away from validating, since it clutters the function and, in some cases, gives a false sense that things are working when they're not. This one's a bit trickier since lack of `postId` for `savePost` is valid and could lead to some bizarre and unexpected consequences (in this case, it'll try to create a post but fail due to lack of content).

If we do validate, the question becomes: do we fail quietly or loudly?